base/tools/src/main/java/com/netscape/cmstools/pkcs12/PKCS12KeyRemoveCLI.java

```diff
@@ -105,7 +106,25 @@ public void execute(CommandLine cmd) throws Exception {
             PKCS12Util util = new PKCS12Util();
 
             PKCS12 pkcs12 = util.loadFromFile(filename, password);
+
+            // find cert that uses the key to be removed
+            PKCS12CertInfo certInfo = pkcs12.getCertInfoByKeyID(keyID.toByteArray());
+
+            if (certInfo != null) {
+                // remove reference to key from cert
+                certInfo.setKeyID(null);
```

Comment:
Maybe at some point we could use a way to put a key back and re-associate it with the cert.